### PR TITLE
feat(analytics): return Group ID with Returned Success

### LINKED ISSUE #890: Add group information to enrollment success event
#873 brought us an explicit event for enrollment success (and failure, and retry).

This ticket seeks to add an event property `payment_group` with the GUID of the Payment Processor group the user was enrolled into.

This will help us track down registration issues, verify users are getting into the correct groups, etc.

## Acceptance Criteria

- [ ] Every time the `enrollment returned` with `status = success` fires, an event property `payment_group` is sent with the event

## Additional context

The Group identifier is stored in the [`EligibilityType` model](https://github.com/cal-itp/benefits/blob/dev/benefits/core/models.py#L45)

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #890 

Passes `eligibility.group_id` as `payment_group` into Amplitude for successful enrollment event

### CODE REVIEW

File: tests/pytest/enrollment/test_views.py [hunk 1]
```diff
@@ -113,15 +113,20 @@ def test_index_eligible_post_valid_form_failure(mocker, client, card_tokenize_fo
...
     mocker.patch("benefits.enrollment.views.api.Client.enroll", return_value=mock_response)
 
     path = reverse(ROUTE_INDEX)
     response = client.post(path, card_tokenize_form_data)
 
-    mocked_analytics_module.returned_success.assert_called_once()
+    mocked_analytics_module.returned_success
     assert response.status_code == 200
     assert response.template_name == TEMPLATE_SUCCESS
     mocked_analytics_module.returned_success.assert_called_once()
```
Comment (machikoyasuda) [inline tests/pytest/enrollment/test_views.py:132]: Why is line 129/132 repeated? I wasn't sure.

I was _trying_ to change the line to something more specific like:

```
mocked_analytics_module.returned_success.assert_called_once_with(app_request_enrollment_success)
```

But I kept getting an error like this:

```python
E       Args:
E       assert (<WSGIRequest...ess'>, '1234') == (<WSGIRequest...nt/success'>,)
E         At index 0 diff: <WSGIRequest: POST '/enrollment/success'> != <WSGIRequest: POST '/enrollment/success'>
E         Left contains one more item: '1234'
E         Full diff:
E         - (<WSGIRequest: POST '/enrollment/success'>,)
E         + (<WSGIRequest: POST '/enrollment/success'>, '1234')
```

and I couldn't figure out how to mock a request that `==` exactly like `(<WSGIRequest: POST '/enrollment/success'>, '1234')`. I would want to make sure that the `1234` group ID is correctly getting passed.

Comment (machikoyasuda) [reply tests/pytest/enrollment/test_views.py]: cc @angela-tran @thekaveman

Comment (thekaveman) [reply tests/pytest/enrollment/test_views.py]: Hmm not sure why that is repeated either, definitely shouldn't be!

Regarding your question, here's an example of checking that a specific call is in the list of calls for the spied function: https://github.com/cal-itp/eligibility-server/blob/main/tests/test_keypair.py#L52

The `mocker` fixture has a helper to create a [Call object](https://docs.python.org/3/library/unittest.mock.html#call), and you can look if that Call is inside the list of all Calls on the spy:

```python
assert mocker.call("arg1", "arg2", "etc.") in spy.call_args_list
```

File: tests/pytest/enrollment/test_views.py [hunk 2]
```diff
@@ -113,18 +113,20 @@ def test_index_eligible_post_valid_form_failure(mocker, client, card_tokenize_fo
...
     mocker.patch("benefits.enrollment.views.api.Client.enroll", return_value=mock_response)
 
     path = reverse(ROUTE_INDEX)
     response = client.post(path, card_tokenize_form_data)
 
-    mocked_analytics_module.returned_success.assert_called_once()
     assert response.status_code == 200
     assert response.template_name == TEMPLATE_SUCCESS
-    mocked_analytics_module.returned_success.assert_called_once()
+    mocked_analytics_module.returned_success.assert_called_once
```
Comment (thekaveman) [inline tests/pytest/enrollment/test_views.py:128]: You still want to call this like a function though: `.assert_called_once()`

**Review by thekaveman — CHANGES_REQUESTED** [review]
Ah, I didn't notice this before:

The `ReturnedEnrollmentEvent` class needs to be updated too, to accept the new parameter (maybe with a default of `None`) and then make a call to `self.update_event_properties()` if there is a value.

**Review by thekaveman — APPROVED** [review]
Nice work! 🚀

**Pull request merged**